src/GmdVizPanel/components/BookmarkHeaderAction.tsx

```diff
@@ -0,0 +1,114 @@
+import { css, cx } from '@emotion/css';
+import { type GrafanaTheme2 } from '@grafana/data';
+import { SceneObjectBase, sceneUtils, type SceneComponentProps, type SceneObjectState } from '@grafana/scenes';
+import { Button, Icon, useStyles2 } from '@grafana/ui';
+import React from 'react';
+
+import { genBookmarkKey } from '../../bookmarks/genBookmarkKey';
+import { notifyBookmarkCreated } from '../../bookmarks/notifyBookmarkCreated';
+import { reportExploreMetrics } from '../../interactions';
+import { PREF_KEYS } from '../../UserPreferences/pref-keys';
+import { userStorage } from '../../UserPreferences/userStorage';
+import { getTrailFor } from '../../utils';
+
+interface BookmarkHeaderActionState extends SceneObjectState {
+  metric: string;
+  isBookmarked: boolean;
+}
+
+export class BookmarkHeaderAction extends SceneObjectBase<BookmarkHeaderActionState> {
+  constructor({
+    metric,
+  }: {
+    metric: BookmarkHeaderActionState['metric'];
+  }) {
+    super({
+      metric,
+      isBookmarked: false,
+    });
+
+    // Update bookmark state when component activates
+    this.addActivationHandler(() => {
+      const actualBookmarkState = this.isCurrentStateBookmarked();
+      this.setState({ isBookmarked: actualBookmarkState });
+    });
+  }
+
+  private isCurrentStateBookmarked(): boolean {
+    try {
+      const trail = getTrailFor(this);
+      const currentUrlState = sceneUtils.getUrlState(trail);
+      const currentKey = genBookmarkKey(currentUrlState);
+      const bookmarksFromStorage = userStorage.getItem(PREF_KEYS.BOOKMARKS) || [];
+      return bookmarksFromStorage.some((b: any) => genBookmarkKey(b.urlValues) === currentKey);
```

Comment:
Just a TS nit, if we export the proper type from [useBookmark.ts](https://github.com/grafana/metrics-drilldown/blob/main/src/bookmarks/useBookmarks.ts#L20): 

```suggestion
      return bookmarksFromStorage.some((b: BookmarkFromStorage) => genBookmarkKey(b.urlValues) === currentKey);
```